### PR TITLE
World name similarity checks

### PR DESCRIPTION
**Pull request by kinshukk**
### Contains

Added checks for similar world names. If a similarly named world is found, a popup is shown.

### How to test

1. Create game -> Advanced Setup -> Universe Setup
2. Add a couple of worlds
3. Rename first to, say, "blocksEverywhere" (by selecting world and going into Configure menu)
4. Try to rename second world to the same name.

### Outstanding before merging

- [ ] Haven't checked for special characters.
- [ ] Didn't quite get that different seed thing mentioned in #3636

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldSetupScreen.java [hunk 1]
```diff
@@ -81,18 +81,48 @@ public void initialise() {
...
+
+                //check for case insensitive equality of world names
+                for (String worldName : universeSetupScreen.worldNames()) {
+                    if (worldName.equalsIgnoreCase(customWorldName.getText())) {
+                        worldNameMatchesAnother = true;
+                        break;
+                    }
+                }
+
+                if (worldNameMatchesAnother == true) {
```
Comment (RatMoleRat) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldSetupScreen.java]: You don't need the `==true` in this if statement - `if (worldNameMatchesAnother) {` does the same thing.

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldSetupScreen.java [hunk 2]
```diff
@@ -81,18 +81,48 @@ public void initialise() {
...
+                    getManager().pushScreen(MessagePopup.ASSET_URI, MessagePopup.class).setMessage("Name Already Used!", "Please use a different name for this world");
+                } else {
+                    //no match found: go back to universe setup
+                    goBack = true;
+                }
+            }
+
+            if (goBack == true) {
```
Comment (RatMoleRat) [inline engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldSetupScreen.java]: Same as above.

File: engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldSetupScreen.java (hunk 1 above)
Comment (kinshukk) [reply engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldSetupScreen.java]: Done :)

Comment (kinshukk) [reply engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldSetupScreen.java]: Checks passing now, artifactory was temporarily down.

**Comment by Cervator** [comment]
@GooeyHub: add to whitelist

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
Tests out fine but I found one sneaky case not covered, and honestly I only just barely thought of it myself: If you add a world, rename it _to its increment + 1_ (as in, rename Perlin-1 to Perlin-2) then add a new world the new world ... is named the same as the prior one you just renamed :-)

@kinshukk are you up for finding where that new name is made up and adding a check there specifically to see if an existing world has been renamed to match the increment, and in that case keep incrementing it until its name is unique?

**Comment by kinshukk** [comment]
Oops yeah I totally missed that one, will look into it.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by shaunakdixit** [comment]
@kinshukk I would like to add one more thing. Don't consider increment of +1 it could be any number suppose if I rename world to **Flat-0** to **Flat-7** and then I add a new world **Flat-8** then You have to print name from **0 (Since 0 is free now we can use it again) to  6 then directly jump to 9**.
I believe best possible solution is to create a list of world number and check that number exist in list if it  is present in it then skip check next number.
https://github.com/MovingBlocks/Terasology/blob/12207a8a2265f44876cc8a94ef095fef404a0c8c/engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java#L277-L285

**Comment by kinshukk** [comment]
That might be ideal, but I think the point of having a world number is just so that the worlds have _unique_ names by default, not for having them in any particular order. Adding more checks for now available world numbers would add unnecessary complexity IMO.

**Comment by shaunakdixit** [comment]
Thats totally up to cervator :-)

**Comment by Cervator** [comment]
Yep for this issue I'm just hunting for world name uniqueness. Good catch tho @shaunakdixit !

Tested out, all looks well, merged! Welcome to the credits section @kinshukk and thanks :D

**Pull request merged**